### PR TITLE
Add iosxr-7.0.2 to nodesets

### PR DESCRIPTION
**Pull request by GomathiselviS**
Signed-off-by: GomathiselviS <gosriniv@redhat.com>

### CODE REVIEW

**Comment by GomathiselviS** [comment]
recheck

**Comment by GomathiselviS** [comment]
recheck

**Comment by GomathiselviS** [comment]
recheck

**Comment by GomathiselviS** [comment]
recheck

File: zuul.d/project.yaml [hunk 1]
```diff
@@ -4,57 +4,10 @@
       - ansible-python-jobs
     check:
       jobs:
-        - ansible-network-asa-appliance:
```
Comment (pabelanger) [inline zuul.d/project.yaml:7]: there is no need to delete these jobs, they will only run is files specific to them change.

**Comment by GomathiselviS** [comment]
recheck

**Comment by GomathiselviS** [comment]
recheck

**Comment by GomathiselviS** [comment]
recheck

**Comment by GomathiselviS** [comment]
recheck

**Comment by GomathiselviS** [comment]
recheck

File: playbooks/ansible-network-iosxr-appliance/files/bootstrap.yaml [hunk 2]
```diff
@@ -21,15 +21,23 @@
...
     # NOTE(pabelanger): This is a huge hack to deal with the fact there is no
     # default gateway seutp.
     - name: Ensure we can SSH into the appliance
       expect:
         command: "ssh admin@{{ hostvars[item].nodepool.private_ipv4 }}"
         responses:
           'Are you sure you want to continue connecting': "yes"
-          'password:': admin
-          'RP/0/0/CPU0:ios#': exit
+          '(?i)password:': admin123
```
Comment (pabelanger) [inline playbooks/ansible-network-iosxr-appliance/files/bootstrap.yaml:39]: we should revert this back to admin, so we don't have to have 2 playbooks for iosxr. This likely means, re-rolling the iosxr images to default to admin.

File: playbooks/ansible-network-iosxr-appliance/files/bootstrap.yaml [hunk 3]
```diff
@@ -43,23 +51,26 @@
       expect:
         command: "sftp admin@{{ hostvars[item].nodepool.private_ipv4 }}"
         responses:
-          'password:': admin
+          '(?i)password:': admin123
           'sftp>':
             - put id_rsa.pub.b64
             - exit
         timeout: 60
+      when: item not in ["iosxr-7.0.2"]
```
Comment (pabelanger) [inline playbooks/ansible-network-iosxr-appliance/files/bootstrap.yaml:59]: lets try not to make this dynamic. if the password is the only issue, lets respin

File: playbooks/ansible-network-iosxr-appliance/files/bootstrap.yaml [hunk 4]
```diff
@@ -21,15 +21,23 @@
   gather_facts: false
   connection: local
   tasks:
+
+    - name: Remove ssh key entry
```
Comment (pabelanger) [inline playbooks/ansible-network-iosxr-appliance/files/bootstrap.yaml:25]: so, we shouldn't need this. As our playbook adds that info on the controller side.

Can you setup an autohold on failure so I can look.  I suspect something in the controller is regenerating the host keys for some reason

File: playbooks/ansible-network-iosxr-appliance/files/bootstrap.yaml (hunk 2 above)
Comment (GomathiselviS) [reply playbooks/ansible-network-iosxr-appliance/files/bootstrap.yaml:39]: @pabelanger iosxr7.0.2 has restrictions on minimum password length which is 6. So we will not be able to use admin. I shall add a check here, for 7.0.2

Comment (pabelanger) [reply playbooks/ansible-network-iosxr-appliance/files/bootstrap.yaml]: k, if that is the case we can move to inventory vars for, which are loaded depending on which image.

**Comment by GomathiselviS** [comment]
recheck

**Comment by GomathiselviS** [comment]
recheck

**Comment by GomathiselviS** [comment]
recheck

**Review by ashwini-mhatre — APPROVED** [review]

File: playbooks/ansible-network-iosxr-appliance/files/bootstrap.yaml [hunk 5]
```diff
@@ -21,15 +21,30 @@
...
+    - set_fact:
+        _iosxr_password: "admin123"
+      when: item in ["iosxr-7.0.2"]
+      with_inventory_hostnames: appliance
+
+    # - name: Remove ssh key generated by Automatic Host Key Generation
+    #  shell: "ssh-keygen -R {{ hostvars[item].nodepool.private_ipv4 }}"
+    #  args:
+    #    executable: /bin/bash
+    #  with_inventory_hostnames: appliance
```
Comment (pabelanger) [inline playbooks/ansible-network-iosxr-appliance/files/bootstrap.yaml:37]: okay, lets add this back for now. Adding a NOTE here to explain what we think is happening.

File: playbooks/ansible-network-iosxr-appliance/files/bootstrap.yaml (hunk 3 above)
Comment (pabelanger) [reply playbooks/ansible-network-iosxr-appliance/files/bootstrap.yaml:66]: lets drop when statement

File: playbooks/ansible-network-iosxr-appliance/files/bootstrap.yaml [hunk 6]
```diff
@@ -60,6 +76,8 @@
     - name: Add SSH public key to zuul user
       iosxr_command:
         commands: "admin crypto key import authentication rsa username zuul disk0:/id_rsa.pub.b64"
+      when: item not in ["iosxr-7.0.2"]
+      with_inventory_hostnames: appliance
```
Comment (pabelanger) [inline playbooks/ansible-network-iosxr-appliance/files/bootstrap.yaml:80]: can we drop when?

File: playbooks/ansible-network-iosxr-appliance/files/bootstrap.yaml [hunk 7]
```diff
@@ -71,3 +89,5 @@
       iosxr_user:
         name: admin
         state: absent
+      when: item not in ["iosxr-7.0.2"]
+      with_inventory_hostnames: appliance
```
Comment (pabelanger) [inline playbooks/ansible-network-iosxr-appliance/files/bootstrap.yaml:93]: same

File: zuul.d/nodesets.yaml [hunk 8]
```diff
@@ -536,6 +535,80 @@
...
+      - name: iosxr-7.0.2
+        label: iosxrv-7.0.2
+    groups:
+      - name: appliance
+        nodes:
+          - iosxr-7.0.2
+      - name: controller
+        nodes:
+          - fedora-32
+
```
Comment (pabelanger) [inline zuul.d/nodesets.yaml:597]: this likely all can remove, we mostly want to test against devel right now.  Given we didn't target our old collections for this version of IOSXR

**Comment by trishnaguha** [comment]
@ashwini-mhatre @Gomathiselvi-S What are we waiting on for the upgrade?

**Comment by ashwini-mhatre** [comment]
> @ashwini-mhatre @Gomathiselvi-S What are we waiting on for the upgrade?

once https://github.com/ansible-collections/cisco.iosxr/pull/130 this PR get approval we can merge this. I already got approval from rohit. waiting for approval from ganesh

**Pull request merged**